pkg/provider/azure_controller_vmssflex.go

```diff
@@ -119,7 +119,7 @@ func (fs *FlexScaleSet) AttachDisk(ctx context.Context, nodeName types.NodeName,
 	klog.V(2).Infof("azureDisk - update(%s): vm(%s) - attach disk list(%+v) returned with %v", nodeResourceGroup, vmName, diskMap, rerr)
 
 	if err == nil && result != nil {
-		if rerr := fs.updateCache(ctx, vmName, result); err != nil {
+		if rerr := fs.updateCache(ctx, vmName, result); rerr != nil {
```

Comment:
good catch, pls make change in another PR, and btw, vmssflex is not supported on aks

Reply:
Will do